pkg/tsdb/prometheus/prometheus.go

```diff
@@ -163,9 +158,25 @@ func parseQuery(queries []*tsdb.Query, queryContext *tsdb.TsdbQuery) (*Prometheu
 		return nil, err
 	}
 
+	intervalFactor := queryModel.Model.Get("intervalFactor").MustInt64(1)
+
+	var step time.Duration
+	stepSeconds, err := queryModel.Model.Get("step").Int64()
```

Comment:
dont think we ever should look at the step model property, it was a bug to use this in the first place and the way it was stored using the last time range you saved the dashboard with makes this pretty random if this is what you actually want.

Reply:
Right, I totally forgot that discussion we had.

Taking a step back I wonder if it makes sense to set step to anything else then the scrape interval for alerting queries. If we scrape every 10s and set step to 1, Prometheus will return the same value every second for 10s. 

Most current reducers wont be affected by this. `count()` and `count_non_null()` would be kinda weird.